### PR TITLE
[Relay] Add test for type inference on generic function

### PR DESCRIPTION
**Pull request by MarisaKirisame**
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/dmlc/tvm/blob/master/CONTRIBUTORS.md#reviewers).

### CODE REVIEW

File: tests/python/relay/test_type_infer.py [hunk 1]
```diff
@@ -193,7 +193,16 @@ def test_equal():
...
     func = relay.Function([i], eq, ret_type=relay.TensorType([], 'int32'))
 
+ 
+def test_dup_type():
+    a = relay.TypeVar("a")
+    av = relay.Var("av", a)
+    make_dup = relay.Function([av], relay.Tuple([av, av]), None, [a])
+    t = relay.TensorType((10,))
+    b = relay.Var("b", t)
+    assert relay.ir_pass.infer_type(make_dup(b)).checked_type == relay.TupleType([t, t])
```
Comment (slyubomirsky) [inline tests/python/relay/test_type_infer.py]: I believe `==` isn't overridden for Types (only exprs) so you need to explicitly call `alpha_equal` here

Comment (MarisaKirisame) [reply tests/python/relay/test_type_infer.py]: it is the opposite - it is overriden for type, not for expr.

Comment (slyubomirsky) [reply tests/python/relay/test_type_infer.py]: Lol

**Comment by MarisaKirisame** [comment]
@tqchen it is a bug

**Comment by tqchen** [comment]
If it is a bug, it would be better to send in a fix along with the regression test.

**Comment by tqchen** [comment]
Closing this for now, let us add the testcase when we upgraded the type inference to support this type of inference.

**Pull request closed without merging**